backend/src/tasks.ts

```diff
@@ -0,0 +1,350 @@
+    .addPositionalParam('payoutAddress', 'reclaim funds to this address')
+    .setAction(async (args, hre) => {
+        await hre.run('compile');
+
+        let quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        const tx = await quiz.reclaimFunds(args.payoutAddress);
+        const receipt = await tx.wait();
+        console.log(`Successfully reclaimed funds to ${args.payoutAddress}. Transaction hash: ${receipt!.hash}`);
+    });
+
+// Set gasless key-pair.
+task('setGaslessKeyPair')
+    .addPositionalParam('address', 'contract address')
+    .addPositionalParam('payerAddress', 'payer address')
+    .addPositionalParam('payerSecret', 'payer secret key')
+    .setAction(async (args, hre) => {
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        const nonce = await hre.ethers.provider.getTransactionCount(args.payerAddress);
+        await setGaslessKeyPair(quiz, args.payerAddress, args.payerSecret, nonce);
+    });
+
+// Fetch image from IPFS.
+task('fetchImageFromIpfs')
+    .addPositionalParam('ipfsHash', 'IPFS hash of the image')
+    .addPositionalParam('outputPath', 'Path to save the fetched image')
+    .setAction(async (args, hre) => {
+        try {
+            const response = await axios.get(`https://ipfs.io/ipfs/${args.ipfsHash}`, {
+                responseType: 'arraybuffer'
+            });
+
+            fs.writeFileSync(args.outputPath, Buffer.from(response.data));
+            console.log(`Image fetched from IPFS and saved to ${args.outputPath}`);
+        } catch (error) {
+            console.error('Error fetching image from IPFS:', error);
+        }
+    });
+
+task('deployAndSetup')
+    .addPositionalParam('configFile', 'Quiz configuration filename',  'config.yaml')
+    .setAction(async (args, hre) => {
+        await hre.run('compile');
+
+        // Load default configuration from YAML file
+        // const defaultConfig = await loadYamlConfig(path.resolve(__dirname, args.configFile));
+        const cfg = await loadYamlConfig(args.configFile);
+
+        console.log(`Questions File: ${cfg.questionsFile}`);
+        console.log(`Coupons File: ${cfg.couponsFile}`);
+        console.log(`Reward: ${cfg.nativeReward}`);
+        console.log(`Fund Amount: ${cfg.fundAmount}`);
+        console.log(`Metadata file path: ${cfg.nftReward.metadataFile}`);
+        console.log(`Fund gasless amount: ${cfg.fundGaslessAmount}`);
+        console.log(`Gasless Address: ${cfg.gaslessAddress}`);
+        console.log(`Gasless Secret: ${cfg.gaslessSecret}`);
+
+
+        const quiz = await hre.run("deployQuiz");
+        await hre.run("addQuestions", { address: quiz, questionsFile: cfg.questionsFile })
+        await hre.run("addCoupons", { address: quiz, couponsFile: cfg.couponsFile })
+        await hre.run("fund", { address: quiz, amount: cfg.fundAmount })
+
+        // Update the contract state if reward > 0
+        if (hre.ethers.parseEther(cfg.nativeReward) > BigInt(0)) {
```

Comment:
nativeReward is optional. This crashes, if you comment it out in the test-config.yaml.

Reply:
Added parameter check for nativeReward.